### PR TITLE
feat: add theme-generator

### LINKED ISSUE #102: 🚀 - theme generator
### Description

reference https://ionicframework.com/docs/theming/color-generator

### PR DESCRIPTION
**Pull request by MishaZhem**
Closes #102

### CODE REVIEW

File: apps/taiga-lumbermill/src/dashboards/theme-generator/theme-example/theme-example.component.html [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    class="example-block"
+>
+    <div tuiHeader="l">
+        <h3 tuiTitle>
+            Title example
+            <div tuiSubtitle>Subtitle example</div>
+        </h3>
+    </div>
+    <tui-input-number
+        tuiHintContent="Dollar sign is commonly placed BEFORE the amount. Use [prefix]."
```
Comment (vladimirpotekhin) [inline apps/taiga-lumbermill/src/dashboards/theme-generator/theme-example/theme-example.component.html:13]: remove this hint

File: apps/taiga-lumbermill/src/dashboards/theme-generator/theme-example/theme-example.component.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        'error',
+        'warning',
+        'neutral',
+        'info',
+    ];
+
+    protected readonly buttons = ['primary', 'accent', 'destructive', 'flat', 'outline'];
+
+    @Input()
+    public host: Signal<string> = signal('');
```
Comment (vladimirpotekhin) [inline apps/taiga-lumbermill/src/dashboards/theme-generator/theme-example/theme-example.component.ts:82]: public theme

File: apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import {ThemeExampleComponent} from './theme-example/theme-example.component';
+import {data} from './theme-generator.constants';
+
+@Component({
+    standalone: true,
+    selector: 'lmb-theme-switcher',
+    imports: [
+        CommonModule,
+        FormsModule,
+        ReactiveFormsModule,
```
Comment (vladimirpotekhin) [inline apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts:32]: ReactiveFormsModule is unused, check others as well

File: apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    TuiIslandDirective,
+    TuiTextfieldControllerModule,
+} from '@taiga-ui/legacy';
+
+import {ThemeExampleComponent} from './theme-example/theme-example.component';
+import {data} from './theme-generator.constants';
+
+@Component({
+    standalone: true,
+    selector: 'lmb-theme-switcher',
```
Comment (vladimirpotekhin) [inline apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts:28]: lmb-theme-generator

File: apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.constants.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
+interface ThemeData {
+    readonly name: string;
+    readonly base_color: string;
```
Comment (vladimirpotekhin) [inline apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.constants.ts:3]: let's use camelCase

Comment (vladimirpotekhin) [reply apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.constants.ts:3]: and it's better
{name, initialValue, variable}

File: apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    templateUrl: './theme-generator.component.html',
+    styleUrl: './theme-generator.component.less',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ThemeGeneratorComponent {
+    protected themeData = data;
+    protected readonly palette = TUI_DEFAULT_INPUT_COLORS;
+    protected colors = this.themeData.map((val) => signal(val.base_color));
+
+    protected host = computed(() =>
```
Comment (vladimirpotekhin) [inline apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts:58]: protected theme

File: apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        TuiTextfield,
+        TuiTextfieldControllerModule,
+        TuiTitle,
+    ],
+    templateUrl: './theme-generator.component.html',
+    styleUrl: './theme-generator.component.less',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ThemeGeneratorComponent {
+    protected themeData = data;
```
Comment (vladimirpotekhin) [inline apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts:54]: no need to store this field

Comment (MishaZhem) [reply apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.ts:54]: themeData is used for input color in html

File: apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.html [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<div class="list-colors">
+    <div tuiHeader="m">
+        <h3 tuiTitle>Colors</h3>
+    </div>
+    @for (InputColor of themeData; track $index) {
```
Comment (vladimirpotekhin) [inline apps/taiga-lumbermill/src/dashboards/theme-generator/theme-generator.component.html:5]: camelCase

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**